tests/shared/src/test/scala/munit/AsyncFixtureSuite.scala

```diff
@@ -0,0 +1,113 @@
+package munit
+
+import scala.concurrent.Future
+import scala.concurrent.Promise
+
+class AsyncFixtureSetupTestTeardownOrderSuite extends FunSuite {
+  val latch = Promise[Unit]
+  var completedFromTest = Option.empty[Boolean]
+  var completedFromTeardown = Option.empty[Boolean]
+
+  val latchOnTeardown = FunFixture.async[String](
+    setup = { test => Future.successful(test.name) },
+    teardown = { name =>
+      implicit val ec = munitExecutionContext
+      Future {
+        completedFromTeardown = Some(latch.trySuccess(()));
+      }
+    }
+  )
+
+  override def afterAll(): Unit = {
+    assertEquals(completedFromTest, Some(true))
+    assertEquals(completedFromTeardown, Some(false))
+  }
+
+  latchOnTeardown.test("test is ran before the teardown") { _ =>
```

Comment:
>> "test is ran before the teardown"
I still struggle with English, but I suggest something like
"test ran before the teardown"
or 
"test has run before the teardown"
or
"test is run before the teardown"

Depends upon the style adopted for the labels of other tests: present or past tense.

Reply:
I actually turned the sentence around and changed it to "teardown runs only after test completes"